### ISSUE TITLE
Cutting off the ending of the sentence 

### ISSUE THREAD

**Issue by MasaStanisavljevic** [CorentinJ/Real-Time-Voice-Cloning#585]
I've noticed that ending of the last word in the text is often cut off. For example, in the output for "This work reflects a quest for lost identity, a recuperation of an unknown past." past is pronounced as pa or pas. I thought that this could be happening because "st" and "t" sounds at the end are registered as noise and trimmed in the post processing step. Upon listening to the outputs without applying the postprocessing operation, I realized that my guess wasn't correct because the "st" and "t" sounds were still missing. I'm using the 002 vocoder from the [list](https://github.com/CorentinJ/Real-Time-Voice-Cloning/issues/400) of available pretrained models for this project. I've used the original synthesizer, synthesizer 004 and synthesizer 006 from the aforementioned list and noticed the same thing happening with all three. Does anyone has any idea why this could be happening or has any advice on how I can further investigate this issue? Thanks!

**Comment by blue-fish** [comment]
It's either the attention mechanism or the stop token prediction. Some users work around this by padding the input with words to be trimmed in post-processing.

You can [return the alignments](https://github.com/CorentinJ/Real-Time-Voice-Cloning/blob/47a9093f901469486fca5b83babeb26ca98fa85d/synthesizer/inference.py#L104) during inference and [plot them](https://github.com/CorentinJ/Real-Time-Voice-Cloning/blob/47a9093f901469486fca5b83babeb26ca98fa85d/synthesizer/utils/plot.py#L15-L38). For the input `Welcome to the toolbox!` I get [this plot](https://user-images.githubusercontent.com/67130644/97897271-18914d00-1ceb-11eb-9655-cbf879d48249.png). The "encoder timestep" refers to the input characters (don't forget to account for [text normalization](https://github.com/CorentinJ/Real-Time-Voice-Cloning/blob/47a9093f901469486fca5b83babeb26ca98fa85d/synthesizer/utils/text.py#L56-L62)), for each "decoder timestep" Tacotron will generate `hparams.outputs_per_step=2` frames of mel spectrogram output (each equal to frame_shift_ms or hop_size/sample_rate, default 0.0125 sec).

By scrutinizing this output you can figure out when Tacotron is working on each letter of your input sequence, and if there's an attention failure towards the end. Since it consistently happens at the end, it's more likely an issue with the stop token being predicted too early. If this is the case, then it could be a problem of the training data or the algorithm itself.

Nowadays attention is being replaced with feedforward schemes where Tacotron is trained to predict phoneme duration which leads to more stable and predictable output from inference. For a repo you can see [ForwardTacotron](https://github.com/as-ideas/ForwardTacotron) and for a paper you can check out [FastSpeech](https://arxiv.org/abs/1905.09263) or [Non-Attentive Tacotron](https://arxiv.org/abs/2010.04301).

I'll leave this open for a few days to give others a chance to comment and discuss, but this is "wontfix" unless you are going to work on it.
